extensions/aris/src/bigDataCluster/controller/controller.ts

```diff
@@ -0,0 +1,83 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { EndpointRouterApi } from './apiGenerated';
+import { IEndPointsResponse, IControllerError, IEndPointsRequest, IHttpResponse, IEndPoint } from './types';
+
+export class ArisController {
+
+	private static adjustUrl(url: string): string {
+		if (!url) {
+			return undefined;
+		}
+
+		url = url.trim().replace(/ /g, '').replace(/,(\d+)$/, ':$1');
+		if (!url.includes('://')) {
+			url = `https://${url}`;
+		}
+		return url;
+	}
+
+	public static async getEndPoints(
+		url: string, username: string, password: string, ignoreSslVerification?: boolean
+	): Promise<IEndPointsResponse> {
+
+		if (!url || !username || !password) {
+			return undefined;
+		}
+
+		url = this.adjustUrl(url);
+		let ep = new EndpointRouterApi(username, password, url);
+		ep.ignoreSslVerification = !!ignoreSslVerification;
+
+		let controllerResponse: IEndPointsResponse = undefined;
+		let controllerError: IControllerError = undefined;
+		let request = <IEndPointsRequest>{
+			url: url,
+			username: username,
+			password: password,
+			method: 'endPointsGet'
+		};
+
+		await ep.endpointsGet().then(result => {
```

Comment:
>then [](start = 26, length = 4)

Instead of awaiting a .then you can just await and then have the code continue after, it's much cleaner (and wrap in a try/catch for the error)

try {
   let result = await ep.endpointsGet();
   controllerResponse = ....
} catch(error) {
   if('response' in error)
       ...
}

Reply:
fixed.